tgui/packages/tgui/interfaces/GasCanister.js

```diff
@@ -90,18 +92,18 @@ export const GasCanister = (props, context) => {
               : <Box color="average">The release valve is missing.</Box> }
           </Fragment>
         </PortableBasicInfo>
-        { !detonator && (<PortableHoldingTank
-          holding={holding}
-          onEjectTank={handleEjectTank} />) }
-        { detonator && (<Detonator
+        { detonator ? (<Detonator
           detonator={detonator}
           detonatorAttachments={detonatorAttachments}
           onToggleAnchor={handleToggleAnchor}
           onToggleSafety={handleToggleSafety}
           onWireInteract={handleWireInteract}
           onPrimeDetonator={handlePrimeDetonator}
           onTriggerActivate={handleTriggerActivate}
-          onSetTimer={handleSetTimer} />) }
+          onSetTimer={handleSetTimer} />)
+          : (<PortableHoldingTank
```

Comment:
Your indenting here is kinda weird. Good rule of thumb for ternary rendering:
```
{
  condition
    ? (
      <ComponentA
        someProp="foo"
        otherProp="bar" /> // (I don't like where the linter requires the `>` to be right now, but it's what we've got
    )
    : (
      <ComponentB
        someProp="foo"
        otherProp="bar" />
    )
}
```
May seem overly newline-y, but makes conditions easier to follow.

Reply:
fixed